geopandas/geodataframe.py

```diff
@@ -1458,7 +1474,17 @@ def _constructor(self):
 
     @property
     def _constructor_sliced(self):
-        return _geoseries_constructor_with_fallback
+        def _geodataframe_constructor_sliced(
+            data=None, index=None, crs=None, *, self_df=self, **kwargs
```

Comment:
I think we can directly use `self` inside the function without passing it to the function? (since it is defined inline here)

Reply:
yep, originally I had this as a standalone function rather than inline function.